src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/HarvestCropsBehavior.java

```diff
@@ -93,15 +91,26 @@ private static boolean isBlockCrops(ItemStack stack, Level world, Player player,
     }
 
     private static boolean harvestBlockRoutine(ItemStack stack, BlockPos pos, Player player) {
-        BlockState blockState = player.level().getBlockState(pos);
-        Block block = blockState.getBlock();
-        CropBlock blockCrops = (CropBlock) block;
-        if (blockCrops.isMaxAge(blockState)) {
-            NonNullList<ItemStack> drops = NonNullList.create();
-            drops.addAll(Block.getDrops(blockState, (ServerLevel) player.level(), pos, null));
-            dropListOfItems(player.level(), pos, drops);
-            player.level().levelEvent(LevelEvent.PARTICLES_DESTROY_BLOCK, pos, Block.getId(blockState));
-            player.level().setBlock(pos, blockCrops.getStateForAge(0), Block.UPDATE_ALL);
+        var level = player.level();
+        var blockState = level.getBlockState(pos);
+        var block = blockState.getBlock();
+        var cropBlock = (CropBlock) block;
+        final var seed = cropBlock.getCloneItemStack(level, pos, blockState).getItem();
+        if (cropBlock.isMaxAge(blockState)) {
+            var drops = Block.getDrops(blockState, (ServerLevel) level, pos, null);
+            for (int i = 0; i < drops.size(); i++) {
+                var drop = drops.get(i);
+                if (drop.is(seed)) {
+                    drop.shrink(1);
+                    if (drop.isEmpty()) {
+                        drops.remove(drop);
```

Comment:
```suggestion
                        drops.remove(i);
```
You already know the index - its the current one, remove via index instead of forcing a search through the list.

Reply:
I switched it to an iterator